R/utils-time.R

```diff
@@ -13,30 +13,21 @@
 #' @importFrom lubridate hours
 #' @export
 flexibleTimeParse <- function(x, timezone, shiftTimeToNoon = TRUE) {
+  format_str <- c("Ymd HMOS z", "Ymd T* z*", "Ymd HMS")
   
-  #first attempt utc
-  format_str <- "Ymd HMOS z"
   time <- parse_date_time(x,format_str, tz=timezone,quiet = TRUE)
   
-  #then attempt an offset time
-  if(isEmptyOrBlank(time)) {
-    format_str <- "Ymd T* z*"
-    time <- parse_date_time(x,format_str, tz=timezone, quiet = TRUE)
-  }
+  dvTimes <- x[which(is.na(time))]
```

Comment:
If you think the sapply is slowing things down, we could just use `is.na`. The times should be returned as `NA` from the `parse_date_time` function, so that could be a sure thing. If that's the case, replace the sapply line with `which.still.NA <- is.na(time)`

I think that is good assumption considering the follow still produce NA:
```r
library(lubridate)
format_str <- c("Ymd HMOS z", "Ymd T* z*", "Ymd HMS")

# empty string
time <- parse_date_time("",format_str, tz="UTC",quiet = TRUE)
time
[1] NA

# NULL
time <- parse_date_time(NULL,format_str, tz="UTC",quiet = TRUE)
time
character(0)

# the vectorized check could be:
which.still.wrong <- is.na(time) & length(time) > 0

# though, that seems to be an issue when any are NULL because it doesn't keep it in the final variable
time <- parse_date_time(c("", NULL, "2017-01-01T12:12:13Z"),format_str, tz="UTC",quiet = TRUE)
time
[1] NA                        "2017-01-01 12:12:13 UTC"

# although neither does isEmptyOrBlank
# not sure if that's a scenario we need to prevent, but it might have to go at the beginning because you would end up returning a vector that is a different length of the original.
```